main.go

```diff
@@ -187,7 +187,7 @@ func wrappedMain() int {
 		// Set this so that we don't get colored output in our machine-
 		// readable UI.
 		if err := os.Setenv("PACKER_NO_COLOR", "1"); err != nil {
-			fmt.Fprintf(os.Stdout, "%s Packer failed to initialize UI: %s\n", ErrorPrefix, err)
```

Comment:
Looking into the  ErrorPrefix redirecting output a bit more I found that in https://github.com/hashicorp/packer/blob/master/main.go#L410 we create a set of io.Readers that will copy output with certain prefixes into the respective output streams. 

The ErrorPrefix meaning that every outputted line following the first occurrence of ErrorPrefix will be copied to stderr. Since the version output does not have a prefix associated with it, and no other lines are outputted to reset the copy to stderr, the copyOutput function will continue to copy its output to stderr. The same issue occurs when running `packer build &`

To validate this I added a log line with the OutputPrefix to the code on master to see if the version output gets copied into the correct output stream.

```go
        fmt.Fprintf(os.Stdout, "%s Running in background, not using a TTY\n", ErrorPrefix)
        fmt.Fprintf(os.Stdout, "%s", OutputPrefix)
```

Results with the change
```bash
~>  ./packer.master version 1>stdout 2>stderr &
[2] 8191

[2]    done       ./packer.master version > stdout 2> stderr

~>  cat stdout
Packer v1.7.2-dev

~>  cat stderr
 Running in background, not using a TTY
```

I see that your change takes a different approach and now uses ui.Error which outputs to stdout. But it makes me wonder if adding more output to stdout will break the workflow for other users.